frame/contracts/src/exec.rs

```diff
@@ -1788,6 +1867,42 @@ mod tests {
 		});
 	}
 
+	#[test]
+	fn origin_returns_proper_values() {
+		let code_charlie = MockLoader::insert(Call, |ctx, _| {
+			// BOB is caller but not the origin of the stack call
+			assert_ne!(ctx.ext.origin(), &BOB);
```

Comment:
I think calling `origin` twice is kind of redundant.

```suggestion
			assert_eq!(ctx.ext.caller(), &BOB);
```